NEWS.md

```diff
@@ -73,6 +73,33 @@ unit = "s")
 
 10. The dimensions of objects in a `list` column are now displayed, [#3671](https://github.com/Rdatatable/data.table/issues/3671). Thanks to @randomgambit for the request, and Tyson Barrett for the PR.
 
+11. `topn(vec ,n=6L, decreasing=FALSE)` implemented in C by Morgan Jacob, [#3804](https://github.com/Rdatatable/data.table/issues/3804), is inspired by `dplyr::top_n`. It returns the top largest or smallest `n` values for a given numeric vector `vec`. Please see `?topn` for more details.
+
+```R
+set.seed(123)
+x = rnorm(5e7) # 382 MB
+microbenchmark::microbenchmark(
+  topn(x, 6L),
+  order(x)[1:6], 
+  times = 10L
+)
+# Unit: seconds
+#          expr    min    lq  mean  median    uq   max neval
+# topn(x, 6L)     0.19  0.19  0.20    0.20  0.20  0.22    10
+# order(x)[1:6]   4.56  4.60  4.65    4.62  4.70  4.77    10
+
+microbenchmark::microbenchmark(
+  x[topn(x, 6L)],
+  sort(x)[1:6], 
```

Comment:
```sort``` has a ```partial``` argument but it returns the full vector which means it still needs to be subsetted. Your implementation is still 6 times faster. 
```

x = rnorm(5e7) 
microbenchmark::microbenchmark(
sort(x, partial = 1:6)[1:6],
x[topn(x, 6L)],
times = 10L
)
##Unit: milliseconds
##                        expr      min       lq     mean   median       uq
## sort(x, partial = 1:6)[1:6] 689.2710 724.5292 730.2857 728.4854 735.3350
##              x[topn(x, 6L)] 128.7904 129.3341 151.5083 134.5046 159.8133

identical(sort(x, partial = 1:6)[1:6],x[topn(x, 6L)])
## [1] TRUE

Reply:
Thanks. I have updated the benchmarch. Your machine is clearly faster than mine :-)

Reply:
TIL about `partial=TRUE`, pretty cool.